support-frontend/assets/helpers/subscriptionsForms/submit.ts

```diff
@@ -148,32 +149,18 @@ function getGiftRecipient(giftingState: GiftingState) {
 function buildRegularPaymentRequest(
 	state: AnyCheckoutState,
 	paymentAuthorisation: PaymentAuthorisation,
+	addresses: Addresses,
+	promotions?: Promotion[],
 	currencyId?: Option<IsoCurrency>,
 ): RegularPaymentRequest {
 	const { title, firstName, lastName, email, telephone } =
 		state.page.checkoutForm.personalDetails;
-	const {
-		billingPeriod,
-		fulfilmentOption,
-		productOption,
-		productPrices,
-		deliveryInstructions,
-		csrUsername,
-		salesforceCaseId,
-		debugInfo,
-	} = state.page.checkout;
-	const addresses = getAddresses(state);
-	const price = getProductPrice(
-		productPrices,
-		addresses.billingAddress.country,
-		billingPeriod,
-		fulfilmentOption,
-		productOption,
-	);
```

Comment:
These `const` have been edited as we're now passing in some of this info as parameters (`addresses` and `promotions`)